src/extra/gfx.cpp

```diff
@@ -21,11 +21,13 @@ struct DrawablePassShard {
 
   static inline Parameters params{
       {"Features", SHCCSTR("Features to use."), {Type::VariableOf(Types::PipelineStepSeq), Types::PipelineStepSeq}},
+      {"Queue", SHCCSTR("The queue to draw from (optional). Uses the default queue if not specified"), {Types::DrawQueue, Type::VariableOf(Types::DrawQueue)}},

```

Comment:
If it's optional, shouldn't it include `CoreInfo::NoneType` as a valid input type?

Reply:
This is about the case where you pass a variable that is not set, right?
I guess there is a case where you dynamically set the variable from a branch ,can add it.

Reply:
That's the case where you don't pass anything (don't even set the parameter) or when you set it to `nil`. I'm not sure if it is enforced (that if None is not part of the accepted type and the parameter is not then the shard validation fails. But maybe it should be. @sinkingsugar).

Reply:
It already kind of works without since it can only be a ContextVariable, ignore the plain Types::DrawQueue, that was a mistake

Reply:
Yeah I thought they were always optional, that seems to be how it works currently (not checked if they're missing or not)

Reply:
From the UX perspective if it's not populated the chain won't run I guess. It's better to have None and explicit fail during compose if still None though. It's something we will have to stabilize once we build a non-code editor anyway.

Reply:
Added NoneType in these cases to the list of parameter types